LanguageFeatures/Patterns/Exhaustiveness/lifting_cast_pattern_A01_t02.dart

```diff
@@ -0,0 +1,58 @@
+// Copyright (c) 2023, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// @assertion The lifted space union for a pattern with matched value type M is
+/// ...
+/// Cast pattern:
+/// The space union spaces for a cast pattern with cast type C is a union of:
+/// - The lifted space union of the cast's subpattern in context C.
+/// - For each space E in the expanded spaces of M:
+///   a. If E is not a subset of C and C is not a subset of M, then the lifted
+///   space union of E.
+///
+/// @description Check a lifted space of a cast pattern in case of a sealed type
+/// @author sgrekhov22@gmail.com
+/// @issue 51877
+
+// SharedOptions=--enable-experiment=patterns,class-modifiers
+
+import "../../../Utils/expect.dart";
+
+sealed class A {
+  final int field;
+  A(this.field);
+}
+class B extends A {
+  B(int field) : super(field);
+}
+class C extends A {
+  C(int field) : super(field);
+}
```

Comment:
I think we should have a `class BC extends B implements C {}` in order to see that nothing breaks if we're calling the test methods with a `BC`.

Reply:
Added